src/common/list-syntax.k

```diff
@@ -0,0 +1,311 @@
+                      | ExceptionTypeList
+                      | IdList
+                      | TypeDecList
+                      | VarDecList
+                      | ImportDecList
+                      | ActualTypeArgList
+                      | ClassOrInterfaceTypeList
+                      | TypeParamList
+                      | AbstractMethodModList
+                      | AnnoAbstractMethodModList
+                      | AnnoMethodModList
+                      | AnnoVarModList
+                      | AnnoClassModList
+                      | AnnoConstrModList
+                      | AnnoConstantModList
+                      | AnnoFieldModList
+                      | AnnoInterfaceModList
+                      | AnnoList
+                      | AnnoElemDecList
+                      | InterfaceMemberDecList
+                      | ElemValPairList
+                      | ElemValList
+                      | StringPartList
+                      | EnumConstList
+                      | ClassBodyDecList
+
+
+                      syntax   InterfaceType
+                      syntax   ExceptionType
+                      syntax   Id
+                      syntax   TypeDec::= ClassDec
+                      syntax   VarDec
+                      syntax   ImportDec
+                      syntax   ActualTypeArg
+                      syntax   ClassOrInterfaceType
+                      syntax   TypeParam
+                      syntax   AbstractMethodMod
+                      syntax   AnnoAbstractMethodMod
+                      syntax   AnnoMethodMod
+                      syntax   AnnoVarMod
+                      syntax   AnnoClassMod
+                      syntax   AnnoConstrMod
+                      syntax   AnnoConstantMod
+                      syntax   AnnoFieldMod
+                      syntax   AnnoInterfaceMod
+                      syntax   Anno
+                      syntax   AnnoElemDec
+                      syntax   InterfaceMemberDec
+                      syntax   ElemValPair
+                      syntax   ElemVal
+                      syntax   StringPart
+                      syntax   EnumConst
+                      syntax   ClassBodyDec
+
+endmodule
+
+module LIST-MACRO
+imports CORE-FUNCTIONS
+
+syntax KItem ::= toKListWrap(K) [function]
+                | toKListWrap(K, KListWrap) [function]
+
+rule toKListWrap(UL) => toKListWrap(UL,[.KList])
+
+rule toKListWrap(Label:KLabel(Hd,,Tl), [L:KList]) => toKListWrap(Tl, [L,,Hd])
+
+rule toKListWrap('.List`{""`}(.KList), KL:KListWrap) => KL
+rule toKListWrap('.List`{"`,"`}(.KList), KL:KListWrap) => KL
+rule toKListWrap('.List`{"."`}(.KList), KL:KListWrap) => KL
+rule toKListWrap('.List`{"&"`}(.KList), KL:KListWrap) => KL
+
+
+//AnnoClassModList "class" Id  TypeParams? Super? Interfaces? -> ClassDecHead {cons("ClassDecHead")}
+rule    'ClassDecHead(('AnnoClassModList(A) => toKListWrap('AnnoClassModList(A))),,_) [anywhere]
```

Comment:
Yes, I changed all [anywhere] to [macro]. So far so good, we'll see if all tests pass.